packages/utils/src/utils/agent.utils.ts

```diff
@@ -42,3 +42,102 @@ export const createAgent = async ({
     shouldFetchRootKey: fetchRootKey,
   });
 };
+
+export interface AgentManagerConfig {
+  fetchRootKey: boolean;
+  host: string;
+}
+
+/**
+ * AgentManager class manages HttpAgent instances for different identities.
+ *
+ * It caches agents by identity to optimise resource usage and avoid unnecessary agent creation.
+ * Provides functionality to create new agents, retrieve cached agents, and clear the cache when needed.
+ */
+export class AgentManager {
+  private agents: Record<string, HttpAgent> | undefined = undefined;
+
+  private constructor(private readonly config: AgentManagerConfig) {
+    this.config = config;
+  }
+
+  /**
+   * Static factory method to create a new AgentManager instance.
+   *
+   * This method serves as an alternative to directly using the private constructor,
+   * making it more convenient to create instances of `AgentManager` using a simple and clear method.
+   *
+   * @param {AgentManagerConfig} config - Configuration options for the AgentManager instance.
+   * @param {boolean} config.fetchRootKey - Whether to fetch the root key for certificate validation.
+   * @param {string} config.host - The host to connect to.
+   * @returns {AgentManager} A new instance of `AgentManager`.
+   */
+  public static create(config: AgentManagerConfig): AgentManager {
```

Comment:
You added a factory to reflects the fact that the Agent itself also uses such a pattern, do I get it right?

Reply:
correct, I wanted to keep it kind of consistent with its sub-elements. Being a "collection" of agents, I would like to see that I can initialize it in the same "pattern"

Reply:
Gotcha, works for me.